### PR TITLE
Remove serviceSetupV2 feature flag.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #4901

## Relevant technical choices

I've spent a wild amount of time trying to get this failing test working: https://github.com/google/site-kit-wp/pull/4950/files#diff-f60fee9e4d0635bf0b187460fdc3ae78bcc0ed97c1bd2f91e7256bc402b5616dR152

It seems the E2E plugins that should simulate a success OAuth sign-in aren't working for this one test. I thought all the other tests were maybe using the GCP sign-in flow, but that doesn't seem to be the case. But this one test always tries to authenticate against the real proxy and returns an error when I `var_dump()` the response:

image

I'm totally lost as the E2E plugins and OAuth code has been tough for me to follow and work backward through. Even using the site manually didn't help much.

If anyone has ideas that'd be great, but I think I need to step back from it as I'm just spinning my wheels. If fresh eyes can fix it that'd be great, but otherwise I've just skipped the tests after manually seeing that the User Input will still work after setup.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @tofumatt – this looks good overall although I noticed one place where the feature flag is still used (may be from the last merge). It's only used in a test but that test is also failing as a result. See `AuthenticationTest::test_filter_features_via_proxy`

There was also one place where a condition unrelated to the flag shouldn't have been removed; see below.

Regarding the issue with User Input + E2E, I'll take a closer look at that now but if we need to leave that skipped for the time being that should be ok since it isn't being rolled out yet anyways. It would be preferable of course to fix instead though 😄

File: includes/Core/Authentication/Authentication.php
Comment (tofumatt) [reply includes/Core/Authentication/Authentication.php:1406]: Argh, yes, I think I did the merge wrong, thanks 👍🏻

File: includes/Core/Authentication/Authentication.php
Comment (aaemnnosttv) [inline includes/Core/Authentication/Authentication.php:1407]: I think the option is still no longer necessary here because the feature is on 100% of the time now. As is, this would enable all features if the user had this option set which is not what we want 😄 

The important thing to keep here is that we don't attempt to fetch remote features if there are no credentials as that request would fail.

I'm about to push the fixes for user input now so I'll fix this as well 👍

Comment (tofumatt) [reply includes/Core/Authentication/Authentication.php]: Cheers, I think after the `main` merge this code sorta confused me 😅

**Comment by aaemnnosttv** [comment]
I fixed the E2E test that was being skipped before for user input. The problem wasn't that it was trying to make a request to the proxy (these are all blocked by a WP constant) but that because it was blocked, there was no `Redirect-To` header in the response which also triggers the same error message that was being shown. In order to provide this, I had to add an e2e plugin to stub the response for this request which made the previous client-side request interception no longer necessary as well. In reality it wasn't quite that simple to figure out and I ran into a really weird bug with the plugin activation utility which is still there (I was able to work around it by changing the order of plugins to activate for now), so this is also a note to address this in the future.

There are still a few E2E-related failures but these appear to be the AMP stability-related issues mentioned earlier. This should be good to go now 👍

**Review by aaemnnosttv — APPROVED** [review]
LGTM 👍

**Pull request merged**
